src/MySTMarkdownCell.tsx

```diff
@@ -29,22 +29,35 @@ export class MySTMarkdownCell
   implements IMySTMarkdownCell
 {
   private _doneRendering = new PromiseDelegate<void>();
+  private _doRendering = false;
+
+  mystOptions: MySTOptionsProvider<StaticNotebook>;
 
   myst: {
     pre?: GenericParent;
     post?: GenericParent;
     node?: HTMLDivElement;
   } = {};
 
-  constructor(options: MarkdownCell.IOptions) {
+  constructor(options: MarkdownCell.IOptions, mystOptions: MySTOptionsProvider<StaticNotebook>) {
     super(options);
+    this.mystOptions = mystOptions;
+
+    // BUG: if renderInput() was attempted from super constructor, now it can be done
+    if (this._doRendering) this.renderInput(null as unknown as Widget);
+    this._doRendering = true;
```

Comment:
Is this because downstream you are calling this constructor?

Do we need this render in here at all? Or can we remove it + the option and explicitly call renderInput when we need to?

Reply:
It's all just a horrible hack because of a bug in the 3.5.x parent class. A cursory look tells me it's fixed in the 3.6.x series. I don't know when it's okay for labextensions (jupyterlab-myst) to start following v3.6 -- the cookiecutter still gives you v3.5. But once the bug is gone, this code can be gone.